### PR TITLE
Move shared .proto files from common/metadata/ to shared_protos/

### PR DESCRIPTION
**Pull request by DavidCurtiss**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~~[ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~

~~[ ] I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?

Moves session.proto and nidevice.proto to a clearer path. (Note that "metadata" has a special meaning for gRPC, so having [only] .proto files in a "metadata" folder is a bit odd.)

Also, adds a README.md describing where to update the files from.

### List issues fixed by this Pull Request below, if any.

None

### What testing has been done?

PR

### CODE REVIEW

File: build/defines.mak [hunk 1]
```diff
@@ -8,8 +8,8 @@ TOX_INI := $(OUTPUT_DIR)/tox.ini
 DRIVER_DIR := $(ROOT_DIR)/src/$(DRIVER)
 METADATA_DIR := $(DRIVER_DIR)/metadata
 METADATA_FILES := $(wildcard $(METADATA_DIR)/*.py)
-COMMON_METADATA_DIR := $(ROOT_DIR)/src/common/metadata
-METADATA_DIRS := $(METADATA_DIR) $(COMMON_METADATA_DIR)
+SHARED_PROTOS_DIR := $(ROOT_DIR)/src/shared_protos
+METADATA_DIRS := $(METADATA_DIR) $(SHARED_PROTOS_DIR)
```
Comment (ni-jfitzger) [inline build/defines.mak:12]: It's a little weird having both a METADATA_DIR variable and a METADATA_DIRS variable.

**Review by ni-jfitzger — APPROVED** [review]

File: build/defines.mak (hunk 1 above)
Comment (DavidCurtiss) [reply build/defines.mak]: Renamed to PROTO_DIRS, since we use it to look for .proto files.

File: src/shared_protos/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Updating
+
+To update this metadata folder, copy the .proto files from <https://github.com/ni/grpc-device/tree/main/source/protobuf>.
```
Comment (dmondrik) [inline src/shared_protos/README.md:3]: Is "metadata folder" still an accurate way to refer to this?

**Review by dmondrik — APPROVED** [review]

File: src/shared_protos/README.md (hunk 2 above)
Comment (DavidCurtiss) [reply src/shared_protos/README.md]: Changed it to say just "this folder".

**Review by marcoskirsch — APPROVED** [review]

**Pull request merged**